library/network/src/modules/SuSEFirewallServices.rb

```diff
@@ -61,6 +54,175 @@ class SuSEFirewallServicesClass < Module
       "description"     => ""
     }
 
+    # Function returns the map of supported (known) services.
+    #
+    # @return [Hash{String => String}] supported services
+    #
+    #
+    # **Structure:**
+    #
+    #
+    #     	$[ service_id : localized_service_name ]
+    #     	$[
+    #     	  "dns-server" : "DNS Server",
+    #         "vnc" : "Remote Administration",
+    #       ]
+    def GetSupportedServices
+      supported_services = {}
+
+      all_services.each do |service_id, service_definition|
+        Ops.set(
+          supported_services,
+          service_id,
+          # TRANSLATORS: Name of unknown service. %1 is a requested service id like nis-server
+          Ops.get_string(
+            service_definition,
+            "name",
+            Builtins.sformat(_("Unknown service '%1'"), service_id)
+          )
+        )
+      end
+
+      deep_copy(supported_services)
```

Comment:
OK I just copied the old code without further modifications. I will do as suggested.